core/lib/mno_enterprise/concerns/models/product_instance.rb

```diff
@@ -0,0 +1,56 @@
+module MnoEnterprise::Concerns::Models::ProductInstance
+  extend ActiveSupport::Concern
+
+  #==================================================================
+  # Included methods
+  #==================================================================
+  # 'included do' causes the included code to be evaluated in the
+  # context where it is included rather than being executed in the module's context
+  included do
+    property :created_at, type: :time
+    property :updated_at, type: :time
+
+    #==============================================================
+    # Constants
+    #==============================================================
+    ACTIVE_STATUSES = [:running, :stopped, :staged, :provisioning, :starting, :stopping, :updating]
+    TERMINATION_STATUSES = [:terminating, :terminated]
+  end
+
+  # TODO: Copied over structure from AppInstance.
+  #       Needs to be filled when doing complete migration from AppInstance to
+  #       ProductInstance
+
+  #==================================================================
+  # Class methods
+  #==================================================================
+  module ClassMethods
```

Comment:
The provision! and terminate! actins may not be required neither and managed via subscriptions